spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalogManager.java

```diff
@@ -0,0 +1,106 @@
+public class GravitinoCatalogManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoCatalogManager.class);
+  private static GravitinoCatalogManager gravitinoCatalogManager;
+
+  private volatile boolean isClosed = false;
+  private HashMap<String, Catalog> gravitinoCatalogs = new HashMap<>();
+  private String metalakeName;
+  private GravitinoClient gravitinoClient;
+
+  private GravitinoCatalogManager(String gravitinoUri, String metalakeName) {
+    this.metalakeName = metalakeName;
+    this.gravitinoClient = GravitinoClient.builder(gravitinoUri).build();
+  }
+
+  // It's not thread safe, and is expected to be invoked only once.
+  public static GravitinoCatalogManager createGravitinoCatalogManager(
+      String gravitinoUrl, String metalakeName) {
+    if (gravitinoCatalogManager != null) {
+      throw new RuntimeException("should not create duplicate GravitinoCatalogManager");
+    }
+    gravitinoCatalogManager = new GravitinoCatalogManager(gravitinoUrl, metalakeName);
+    return gravitinoCatalogManager;
+  }
+
+  public static GravitinoCatalogManager getGravitinoCatalogManager() {
+    if (gravitinoCatalogManager == null) {
+      throw new RuntimeException("GravitinoCatalogManager has not created yet");
+    }
+    if (gravitinoCatalogManager.isClosed) {
+      throw new RuntimeException("GravitinoCatalogManager " + "is already closed");
+    }
+    return gravitinoCatalogManager;
+  }
+
+  public void close() {
+    Preconditions.checkArgument(isClosed == false);
+    isClosed = true;
+    gravitinoClient.close();
+  }
+
+  public Catalog getGravitinoCatalogInfo(String name) {
+    return gravitinoCatalogs.get(name);
+  }
+
+  public Map<String, Catalog> getGravitinoCatalogs() {
+    return ImmutableMap.copyOf(gravitinoCatalogs);
+  }
+
+  public String getMetalakeName() {
+    return metalakeName;
+  }
+
+  public void loadCatalogsFromGravitino() {
+    LOG.info("Load metalake: " + metalakeName);
+    GravitinoMetaLake metalake =
+        gravitinoClient.loadMetalake(NameIdentifier.ofMetalake(metalakeName));
+    loadCatalogs(metalake);
+  }
+
+  private void loadCatalogs(GravitinoMetaLake metalake) {
+    NameIdentifier[] catalogNames = metalake.listCatalogs(Namespace.ofCatalog(metalake.name()));
+    LOG.info(
+        "Load metalake {}'s catalogs. catalogs: {}.",
+        metalake.name(),
+        Arrays.toString(catalogNames));
+
+    Arrays.stream(catalogNames)
+        .forEach(
+            (NameIdentifier nameIdentifier) -> {
+              try {
+                String catalogName = nameIdentifier.name();
+                Catalog catalog = metalake.loadCatalog(nameIdentifier);
+                gravitinoCatalogs.put(catalogName, catalog);
+                LOG.info(
+                    "Load catalog {} in metalake {} successfully.", catalogName, metalake.name());
+              } catch (Exception e) {
+                // the catalog maybe not used by SQL, delay the error to SQL analysis phase
```

Comment:
What's the meaning of "delay the error"?

Reply:
GravitinoCatalogManager poll catalogs from Gravitino just once when startup unlike trino because it's short-lived.   If pulling one catalog failed, it continues to run the SQL not quit because the SQL may not use this catalog.

Reply:
Now I prefer to make it poll catalog info periodly to fix the above error.

Reply:
Poll catalog info periodly is not practicable because catalog information is passed though `SQLConf` from GravitinoCataglogManager to Spark CatalogManager.  When Spark CatalogManager startup it will clone a `SQLConf` which makes the changes take no effect.  
One other solution is make GravitinoCatalog lazy load,
1. GravitinoCatalogManager just list all catalogs load all catalogs 
2. GravitinoCatalogManager register GravitinoCatalog to SparkCatalogManager.
3. When using specify catalog, `GravitinoCatalog` lazy load to get catalog information from gravitino, if failed, throw a exception.
The main problem of this solution is we just provide GravitinoCatalog with all capacities like `ViewCatalog` `FunctionCatalog`, and `ProcedureCatalog`.  If using specify catalog like GravitionHiveCatalog, it just implements the real capacities like just implement `FunctionCatalog`.

Overall I prefer the current implement, it's a design choose, @jerryshao @qqqttt123  what do you think?

Reply:
About the solution you mentioned here, how do you achieve this?

Say if you configure a Gravitino catalog called "g" in SQLConf, but the real catalog could be "a", "b", "c". When the user issue a query like "select * from a.db.table", How does Spark know that this catalog "a" belongs to Gravitino catalog "g"?

Reply:
GraivitinoCatalog has the same name with catalog in spark.   Gravitino catalog called "g" will be registered to "g" in Spark.